### PR TITLE
Apply black

### PR DESCRIPTION
**Pull request by dalf**
## What does this PR do?

This PR is demo of [black](https://pypi.org/project/black/):
* first commit `black -l 120 -S searx searxng_extra` : reformat except the simple quote.
* second commit : `black -l 120 searx searxng_extra`: reformat including the simple quote.

## Why is this change important?

Normalize code format.

## Author's checklist

* About PEP8 E203: see https://github.com/psf/black/issues/315
* If we use black, we can disable pycodestyle, and just reformat the code (and keep pylint). The tooling is not modified, so `make test` fails for now.
* there is an emacs package: https://github.com/wbolster/emacs-python-black

### CODE REVIEW

**Comment by return42** [comment]
I started to implement https://github.com/return42/searxng/commits/black-code-style 
/ but I struggle when adding a `pyproject.toml` file.

A `pyproject.toml` breaks our python installation procedure with a misleading error message ... I don't know why users of "black" are forced to change their build stack just to configure a tool .. I gave up.

Here is the error message, pyyaml is installed in the virtualenv, but with a `pyproject.toml`, pip reports that a yaml installation is missed .. very confusing.

```
$ make buildenv
...
BUILDENV  PYENV     [install] pip install -e 'searx[test]'
BUILDENV  Obtaining file:///x/share/searxng
BUILDENV    Installing build dependencies: started
BUILDENV    Installing build dependencies: finished with status 'done'
BUILDENV    Checking if build backend supports build_editable: started
BUILDENV    Checking if build backend supports build_editable: finished with status 'done'
BUILDENV    Getting requirements to build wheel: started
BUILDENV    Getting requirements to build wheel: finished with status 'error'
BUILDENV    ERROR: Command errored out with exit status 1:
BUILDENV     command: /x/share/searxng/local/py3/bin/python /x/share/searxng/local/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpkffhqb23
BUILDENV         cwd: /x/share/searxng
BUILDENV    Complete output (22 lines):
BUILDENV    Traceback (most recent call last):
BUILDENV      File "/x/share/searxng/local/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
BUILDENV        main()
BUILDENV      File "/x/share/searxng/local/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
BUILDENV        json_out['return_val'] = hook(**hook_input['kwargs'])
BUILDENV      File "/x/share/searxng/local/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
BUILDENV        return hook(config_settings)
BUILDENV      File "/tmp/pip-build-env-g39jipud/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 162, in get_requires_for_build_wheel
BUILDENV        return self._get_build_requires(
BUILDENV      File "/tmp/pip-build-env-g39jipud/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 143, in _get_build_requires
BUILDENV        self.run_setup()
BUILDENV      File "/tmp/pip-build-env-g39jipud/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 267, in run_setup
BUILDENV        super(_BuildMetaLegacyBackend,
BUILDENV      File "/tmp/pip-build-env-g39jipud/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 158, in run_setup
BUILDENV        exec(compile(code, __file__, 'exec'), locals())
BUILDENV      File "setup.py", line 7, in <module>
BUILDENV        from searx.version import VERSION_TAG, GIT_URL
BUILDENV      File "/x/share/searxng/searx/__init__.py", line 12, in <module>
BUILDENV        import searx.settings_loader
BUILDENV      File "/x/share/searxng/searx/settings_loader.py", line 8, in <module>
BUILDENV        import yaml
BUILDENV    ModuleNotFoundError: No module named 'yaml'
BUILDENV    ----------------------------------------
BUILDENV  WARNING: Discarding file:///x/share/searxng. Command errored out with exit status 1: /x/share/searxng/local/py3/bin/python /x/share/searxng/local/py3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpkffhqb23 Check the logs for full command output.
```

**Comment by dalf** [comment]
pyaml is required to run setup.py, so it has be declared as such in project.toml:
* ref : https://www.python.org/dev/peps/pep-0517/ and https://www.python.org/dev/peps/pep-0518/
* for example : https://github.com/aio-libs/aiohttp/blob/bb3aead43b0bf9ab9e7e69057cc93865f3d02194/pyproject.toml#L1-L4

But to make it simple why not run black with parameters:
`black -l 120 --target-version py36`
?

black is not very configurable, but as described, it would solve the code formatting discussion.

**Comment by return42** [comment]
> black is not very configurable, but as described, it would solve the code formatting discussion.

Personally I prefer a config file .. what can be read by other tools also. I don't know if we should reinvent our "requirements" setup to have such a toml file .. ?

**Comment by not-my-profile** [comment]
How about using `--skip-string-normalization`? I don't think that string normalization adds much ... rather it needlessly produces a huge diff obscuring `git blame` ... (yes I know you can use `git blame --skip-rev` but normally you only use that after seeing that the autoformat commit changed the line you are interested in). So I think `--skip-string-normalization` would improve git blame ergonomics.

That is how about omitting the second commit?

**Comment by dalf** [comment]
I pushed a new version:

A first commit update `./manage`:
* `make test.pep8` is replaced by `test.black`: black runs on ALL python files including linted by pylint.
* `make format.python` calls black to format the Python code

A second commit contains the result of `make format.python`

A third one fixes some pylint issues (it can be improved, but at least the tests pass).

**Comment by return42** [comment]
> I pushed a new version 

fine, thanks .. I will have a look.

What do you think about `pyproject.toml` in our project, should I give it a try? If so I would take time for a research (its new to me).

[EDIT] now I see there is a `--config` option / no need to have a `pyproject.toml` ..

File: searx/search/checker/impl.py [hunk 1]
```diff
@@ -25,13 +25,87 @@
...
+    'optgroup',
+    'option',
+    'output',
+    'progress',
+    'select',
+    'textarea',
+    'applet',
+    'frame',
+    'frameset',
 ]
```
Comment (return42) [inline searx/search/checker/impl.py]: Over all I think black formatting is good, but there are some exceptions like this one here.

In such cases we should use `# fmt: off` & `# fmt: on`.

**Comment by return42** [comment]
FYI: I am in review with this PR, I will push some suggestions, please do not change the code while I am in review / thanks!

**Review by return42 — APPROVED** [review]
LGTM / thanks!

@dalf See my comment and the patches I pushed on top of your branch.

**Comment by return42** [comment]
We should prioritize (merge) this PR over:

- https://github.com/searxng/searxng/pull/614
- https://github.com/searxng/searxng/pull/613

@dalf if you want I can rebase this branch, but first have a look at the commits I pushed on top of your branch.

**Comment by dalf** [comment]
> first have a look at the commits I pushed on top of your branch.

* c699eb8bd9e58821fa65d2ce1ac7fbbea4f8a598 : okay, why not.
* 14800581f294907d640010109d09a134341f16fb : 👍 

> if you want I can rebase this branch

it will be never a good time to change a lot of time files. So let's go now.

**Comment by return42** [comment]
> it will be never a good time to change a lot of time files. So let's go now.

OK, I will going to rebase this branch right now, please do not merge other PR with python code in the meantime ..

**Comment by return42** [comment]
@dalf I rebased and reorder the branch / If you think PR is ready to merge, please merge.

FYI: the CI hangs for some reason I don't know, I restarted the jobs .. lets see if the tests pass now ...

**Comment by return42** [comment]
Seems `./manage node.env` does not work in the CI ([see here](https://github.com/searxng/searxng/runs/4640940281?check_suite_focus=true#step:4:23)) / not related to this PR I think.

**Comment by dalf** [comment]
Thank you for the update.

**Pull request merged**